airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/Operation.kt

```diff
@@ -11,7 +11,16 @@ import io.airbyte.protocol.models.v0.AirbyteMessage
  * OperationType}s and proxies to an {@link OperationExecutor} that performs the actual work.
  */
 interface Operation {
```

Comment:
Why don't we have a `requiresState` method here?

Reply:
@akashkulk Not sure.  If it makes sense to have each operation implementation declare that it needs the state or not, we can add it.  It's really hard to capture all of the functionality for operations and where the line is between the operation interface and logic specific to the various implementations in the connector.  This is a best effort/first pass at actually defining that interface to separate the command execution from the actual logic of the command.